metrics/ci_integrations/test/integration/ci/ci_integration_test.dart

```diff
@@ -2,242 +2,115 @@
 
+    final firstSyncStage = _SyncStageMock();
+    final secondSyncStage = _SyncStageMock();
+    final syncStages = [firstSyncStage, secondSyncStage];
+    final ciIntegration = CiIntegration(stages: syncStages);
+
+    tearDown(() {
+      reset(firstSyncStage);
+      reset(secondSyncStage);
+    });
+
     test(
-      "throws an ArgumentError if the given source client is null",
+      "throws an ArgumentError if the given stages is null",
       () {
-        expect(
-          () => CiIntegration(
-            sourceClient: null,
-            destinationClient: DestinationClientStub(),
-          ),
-          throwsArgumentError,
-        );
+        expect(() => CiIntegration(stages: null), throwsArgumentError);
       },
     );
 
     test(
-      "throws an ArgumentError if the given destination client is null",
+      "creates an instance with the given sync stages",
       () {
-        expect(
-          () => CiIntegration(
-            sourceClient: SourceClientStub(),
-            destinationClient: null,
-          ),
-          throwsArgumentError,
-        );
+        final ciIntegration = CiIntegration(stages: syncStages);
+
+        expect(ciIntegration.stages, equals(syncStages));
       },
     );
 
     test(
       ".sync() throws an ArgumentError if the given config is null",
       () {
-        final ciIntegration = CiIntegration(
-          sourceClient: SourceClientStub(),
-          destinationClient: DestinationClientStub(),
-        );
-
         expect(() => ciIntegration.sync(null), throwsArgumentError);
       },
     );
 
     test(
-      ".sync() returns an error if a source client throws fetching all builds",
+      ".sync() calls the given sync stages in the right order with the given sync config",
       () async {
-        final sourceClient = SourceClientStub(
-          fetchBuildsCallback: (_) => throw UnimplementedError(),
-        );
-        final destinationClient = DestinationClientStub(
-          fetchLastBuildCallback: (_) => null,
-        );
-        final ciIntegration = CiIntegration(
-          sourceClient: sourceClient,
-          destinationClient: destinationClient,
-        );
-        final result = await ciIntegration.sync(syncConfig);
+        when(firstSyncStage.call(syncConfig)).thenSuccessWith(null);
+        when(secondSyncStage.call(syncConfig)).thenSuccessWith(null);
 
-        expect(result.isError, isTrue);
-      },
-    );
+        await ciIntegration.sync(syncConfig);
 
-    test(
-      ".sync() returns an error if a source client throws fetching the builds after the given one",
-      () async {
-        final sourceClient = SourceClientStub(
-          fetchBuildsAfterCallback: (_, __) => throw UnimplementedError(),
-        );
-        final ciIntegration = CiIntegrationStub(sourceClient: sourceClient);
-        final result = await ciIntegration.sync(syncConfig);
-
-        expect(result.isError, isTrue);
+        verifyInOrder([
```

Comment:
It appears that the `verifyInOrder` does not ensure the given mocks are called once (function [doc](https://pub.dev/documentation/mockito/latest/mockito/verifyInOrder.html)). Should we maybe rewrite this test case or create a new one that ensures all stages are called only once?